.golangci.yml

```diff
@@ -1,87 +1,83 @@
-    local-prefixes: github.com/kopia/kopia
-  govet:
-    shadow: true
-    settings:
-      printf:
-        funcs:
-          - (*go.uber.org/zap.SugaredLogger).Debugf
-          - (*go.uber.org/zap.SugaredLogger).Infof
-          - (*go.uber.org/zap.SugaredLogger).Warnf
-          - (*go.uber.org/zap.SugaredLogger).Errorf
-  lll:
-    line-length: 256
-  loggercheck:
-    zap: true
-    no-printf-like: true
-    rules:
-      - (*go.uber.org/zap.SugaredLogger).With
-  maligned:
-    suggest-new: true
-  misspell:
-    locale: US
-
+version: "2"
 linters:
-  enable-all: true
+  settings:
+    cyclop:
+      max-complexity: 20
+    exhaustive:
+      # indicates that switch statements are to be considered exhaustive if a
+      # 'default' case is present, even if all enum members aren't listed in the
+      # switch
+      default-signifies-exhaustive: true
+    forbidigo:
+      forbid:
+        - pattern: filepath.IsAbs # use ospath.IsAbs which supports windows UNC paths
+        - pattern: ioutil.Discard # use io.Discard
+        - pattern: ioutil.NopCloser # use io.NopCloser
+        - pattern: ioutil.ReadAll # use io.ReadAll
+        - pattern: ioutil.ReadDir # use os.ReadDir
+        - pattern: ioutil.ReadFile # use os.ReadFile
+        - pattern: ioutil.TempDir # use os.MkdirTemp
+        - pattern: ioutil.TempFile # use os.CreateTemp
+        - pattern: ioutil.WriteFile # use os.WriteFile
+        - pattern: time.Now # do not use outside of 'clock' and 'timetrack' packages use clock.Now or timetrack.StartTimer
+        - pattern: time.Since # use timetrack.Timer.Elapsed()
+        - pattern: time.Until # never use this
+        - pattern: Envar\(\" # do not use envar literals, always wrap with EnvName()
+    funlen:
+      lines: 100
+      statements: 60
+    gocognit:
+      min-complexity: 40
+    goconst:
+      min-len: 5
+      min-occurrences: 3
+    gocritic:
+      settings:
+        hugeParam:
+          # size in bytes that makes the warning trigger (default 80)
+          sizeThreshold: 160
+      enabled-tags:
+        - diagnostic
+        - experimental
+        - opinionated
+        - performance
+        - style
+      disabled-checks:
+        - whyNoLint
+        - wrapperFunc
+    gocyclo:
+      min-complexity: 15
+    govet:
+      disable:
+        - fieldalignment
+      enable-all: true
+      settings:
+        printf:
+          funcs:
+            - (*go.uber.org/zap.SugaredLogger).Debugf
+            - (*go.uber.org/zap.SugaredLogger).Infof
+            - (*go.uber.org/zap.SugaredLogger).Warnf
+            - (*go.uber.org/zap.SugaredLogger).Errorf
+    lll:
+      line-length: 256
+    loggercheck:
+      zap: true
+      no-printf-like: true
+      rules:
+        - (*go.uber.org/zap.SugaredLogger).With
+    misspell:
+      locale: US
+
+  default: all
   disable:
+    - gochecknoglobals
     - depguard
     - exhaustruct
-    - exportloopref
+    - exptostd
+    - forcetypeassert
+    - funcorder
```

Comment:
can we keep "exposed" and "funcorder" as enabled? Could be next PR, but maintaining consistency would be great.

Reply:
> I have enabled them but exluded them through `linters.exclusion.rules`.

Please revert that change, it makes it harder to track which linters are disabled.
The disabled linters should be included in this list.

IIUC, both `exptostd` and `funcorder` are new linters. It is OK to keep them disabled for now and enable them separately.

`funcorder` will require a large number of changes, and it is not clear (yet) how much benefit it will provide, if any.